plugins/pulp_rpm/plugins/importers/yum/existing.py

```diff
@@ -135,14 +141,15 @@ def check_all_and_associate(wanted, conduit, config, download_deferred, catalog)
                 # package file does not exist and downloading is not deferred.
                 if not download_deferred and not file_exists:
                     continue
-            catalog.add(unit, wanted[unit.unit_key_as_named_tuple].download_path)
-            if rpm_parse.signature_enabled(config):
-                try:
-                    rpm_parse.filter_signature(unit, config)
-                except PulpCodedException as e:
-                    _LOGGER.debug(e)
-                    continue
-            repo_controller.associate_single_unit(conduit.repo, unit)
+            if unit.id not in all_associated_units:
+                catalog.add(unit, wanted[unit.unit_key_as_named_tuple].download_path)
```

Comment:
@jortel can you help confirm that nothing was depending on the previous behavior? Before this PR, this ``catalog.add()`` call was being made on every sync regardless of whether the package already existed in the repo.

Are there edge cases this impacts? For example, what about transitioning from immediate to on_demand? Or anything else?

I think we definitely want to avoid making this call for every unit on every sync. I'm asking this mostly to make sure that if something else depends on the previous behavior, we find a way to work around that.